docs/threat_models/smart_contracts/multipool_autoswap/mpas_component.puml

```diff
@@ -0,0 +1,83 @@
+@startuml
+'https://plantuml.com/component-diagram
+header Agoric Operating Systems
+footer v0.0.1
+title Multiswap Auto Pool Component Diagram
+
+skinparam componentStyle rectangle
+skinparam defaultTextAlignment center
+
+actor "Multipool Autoswap Contract creator" as cc1
+actor "Random Pool Creator" as cc2
+actor "Person wanting to swap" as cc3
+interface "startInstance()" as si0
+
+package "Multipool Autoswap" {
+    [creatorFacet] as cf0
+    node contractFacet <<zcf>> as conf1 {
+        package "poolSeatA" as pa1 {
+           interface "incrementBy" as pa1i1
+           interface "decrementBy" as pa1i2
+        }
+
+        package "poolSeatB" as pa2 {
+            interface "incrementBy" as pa2i1
+            interface "decrementBy" as pa2i2
+        }
+        package "UserSeat" as use {
+            interface "incrementBy" as usi1
+            interface "decrementBy" as usi2
+        }
+
+        interface "reallocate" as reall
+
+    }
+    package "publicFacet" as pf0 {
+        interface "addPool()" as ap
+        interface "otherMethods..." as om
+        interface "offer()" as offer
+    }
+    si0 -> cf0
+
+
+    node "Pool B" as poolb {
+        [Central Token] as ct2
+        database "Price Registry" <<Central/B>> as pr2
+        [Secondary Token B]
+        [Secondary Token B] --> ct2
+        interface "getPoolSeat()" as gps2
+    }
+
+    node "Pool A" as poola {
+        [Central Token] as ct1
+        database "Price Registry" <<Central/A>> as pr1
+        [Secondary Token A]
+        [Secondary Token A] --> ct1
+        interface "getPoolSeat()" as gps1
+    }
+
+    node "pool n" as pooln {
+    }
+
+    ap -d-> pooln
+
+    offer --> conf1: execute method
```

Comment:
In regards to Zoe, I wanted to show that it referenced the contract and in invoked the contract functionality, however I may have it backwards.  What is the best way to represent `offer()` does stuff with the respective seats here?

Reply:
Here's a quick sketch. Zoe holds funds in escrow, and has the user seats. each user seat is paired with a ZCF seat in the corresponding ZCF vat.

A ZCF vat runs a single contract (in this case MPAS). MPAS has multiple pools, each of which knows two balances that reflect a claim on escrowed funds in Zoe.

When a seat is created for a transaction (`addLiquidity` or `adjustBalances` are the two examples here) it interacts with the conctract code, reallocations are done, and the user gets something back. 

The publicFacet, creatorFacet, etc. could also be pictured within the ZCF vat, since they're facets of the contract.

![Threat Model--multipool components](https://user-images.githubusercontent.com/13372636/135669024-d6788de2-ed84-4c61-b09f-5b463d8bdf33.jpeg)

We can jump on a call if you want to wave my hands and point at the parts of the diagram.